cli/formatter/format.go

```diff
@@ -0,0 +1,800 @@
+func handleColWidth(colAmount int, t table.Writer, opts *Opts) {
+	colWidthTransformer := text.Transformer(func(val interface{}) string {
+		str := fmt.Sprintf("%v", val)
+		widthMax := opts.ColWidthMax
+		if utf8.RuneCountInString(str) > widthMax {
+			firstLine := string([]rune(str)[:widthMax])
+			remainingLines := string([]rune(str)[widthMax:])
+			firstLine = firstLine + "+"
+			remainingLines = text.InsertEveryN(remainingLines, '+', widthMax-1)
+			return firstLine + remainingLines
+		}
+		return fmt.Sprintf("%v", val)
+	})
+	var columnConfigs []table.ColumnConfig
+	for i := 1; i <= colAmount; i++ {
+		columnConfigs = append(columnConfigs,
+			table.ColumnConfig{
+				Number:      i,
+				Transformer: colWidthTransformer,
+				WidthMax:    opts.ColWidthMax,
+			},
+		)
+	}
+	t.SetColumnConfigs(columnConfigs)
+}
+
+// constructMarkdownTable performs construct table in markdown notation.
+func constructMarkdownTable(markdownTable []string, colAmount int,
+	rowsAmount int, opts *Opts) string {
+	if !opts.TransposeMode {
+		markdownTable = insertStrToStrSlice(markdownTable, 0,
+			constructMarkdownEmptyRow(colAmount))
+		markdownTable = insertStrToStrSlice(markdownTable, 1,
+			constructMarkdownHeaderSeparatorRow(colAmount))
+	} else {
+		markdownTable = insertStrToStrSlice(markdownTable, 0,
+			constructMarkdownEmptyRow(rowsAmount))
+		markdownTable = insertStrToStrSlice(markdownTable, 1,
+			constructMarkdownHeaderSeparatorRow(rowsAmount))
+	}
+	var res string
+	for _, v := range markdownTable {
+		res = res + v + "\n"
+	}
+
+	return res
+}
+
+// encodeToCellJsonVal performs encoding as json values for table cell.
+func encodeToCellJsonVal(rowColVal interface{}) (string, error) {
+	jsonVal, err := json.Marshal(convertMapIntrfToMapStr(rowColVal))
+	if err != nil {
+		return "", err
+	}
+
+	return string(jsonVal), nil
+}
+
+// renderMapsWithEqualKeys returns maps with equal keys as single table string.
+func renderMapsWithEqualKeys(maps []map[string]renderNode, opts *Opts) (string, error) {
+	var commonKeys []string
+	for mapKey := range maps[0] {
+		commonKeys = append(commonKeys, mapKey)
+	}
+
+	sort.Strings(commonKeys)
+
+	t := table.NewWriter()
+	var rows []table.Row
+	rows = append(rows, constructHeaderRowForMap(commonKeys))
+
+	for _, mapVal := range maps {
+		var rowVals table.Row
+		for _, commonKey := range commonKeys {
+			if getRenderNodeType(mapVal[commonKey]) == scalarRenderNodeType {
+				switch mapVal[commonKey].(type) {
+				case float64, float32:
+					rowVals = append(rowVals,
+						strconv.FormatFloat(mapVal[commonKey].(float64), 'f', -1, 64))
+				case nil:
+					rowVals = append(rowVals, "nil")
+				default:
+					rowVals = append(rowVals, fmt.Sprintf("%v", mapVal[commonKey]))
+				}
+			} else {
+				// Encoding as json value for map/array table cell cases.
+				jsonVal, err := encodeToCellJsonVal(mapVal[commonKey])
+				if err != nil {
+					return "", err
+				}
+				rowVals = append(rowVals, jsonVal)
+			}
```

Comment:
The duplication is still be here.

Reply:
Done, moved to `encodeScalarNodeToTableRow`